risc0/zkvm/src/host/client/prove/mod.rs

```diff
@@ -120,6 +121,32 @@ pub struct ProverOpts {
     // moment if there is a better place. At some point before 1.0, this option should be moved or
     // dropped.
     pub prove_guest_errors: bool,
+    /// The kind of receipt to be generated.
+    pub receipt_format: ReceiptFormat,
+}
+
+/// This enum represents the various receipt formats that can be generated.
+#[derive(Clone, Serialize, Deserialize, PartialEq, ValueEnum)]
+pub enum ReceiptFormat {
+    /// Composite Receipts resulting from proving a session
+    Composite = 0,
+    /// Succinct Receipts resulting from applying the lift and join programgs so Composite receipts
+    Succinct = 1,
+    /// Compact Receipts resulting from stark2snark on the Succinct receipts
+    Compact = 2,
+}
+
+impl TryFrom<i32> for ReceiptFormat {
+    type Error = anyhow::Error;
+
+    fn try_from(value: i32) -> Result<Self> {
+        match value {
+            0 => Ok(ReceiptFormat::Composite),
+            1 => Ok(ReceiptFormat::Succinct),
+            2 => Ok(ReceiptFormat::Compact),
+            _ => bail!("Unknown receipt format number: {value}"),
+        }
+    }
 }
```

Comment:
I think we should remove this and only have conversions for protobuf over in the `api/convert.rs`